### ISSUE TITLE
Boost-Locale seems not to be auto-linked ICU when updated to 1.81.0

### ISSUE THREAD

**Issue by YunchengLiu** [microsoft/vcpkg#28478]
**Describe the bug**

I use boost-locale with icu backend in my project, everything seems to work well, but when i update boost to 1.81.0. The parts of my project that used character set conversion in boost-locale all crashed and couldn't work properly. I'm pretty sure the problem happened after I updated boost to version 1.81, before that everything was working fine.

After the update, I found that although I installed the boost-locale[icu] option, the icu library file was missing in the generated item (in the previous version, vcpkg would automatically deploy the appropriate library file for me), and I also checked I checked the configuration of vcpkg and my Visual Studio 2022, and confirmed that other libraries work without problems.

**Environment**
- OS:  Windows 11 21H2
- Compiler: MSVC 17.4.3

**Re-Producted**

A small re-producted code is shown below. (C++20)

``` c++
#include <string>
#include <boost/locale.hpp>

int main() {
    auto mgr = boost::locale::localization_backend_manager::global();
    mgr.select("icu");
    auto const input = u8"test";
    auto const conv_input = reinterpret_cast<char const*>(input);
    auto const u16_str = boost::locale::conv::between(conv_input, "UTF-16", "UTF-8", boost::locale::conv::stop);
}

```
I found that in <boost\locale\encoding\codepage.cpp> file, it seems that macro BOOST_LOCALE_WITH_ICU is not defined? so `between` will throw exception anyway.
![image](https://user-images.githubusercontent.com/40971483/208916473-42938f9b-b82a-4557-97ac-34265d4380d4.png)

What should I do to make this problem more concrete and fix it? I depend on boost-1.81 version of other components, so I don't want to rollback the version.

**Comment by MonicaLiu0311** [comment]
@quyykk 
I found your latest [update](https://github.com/microsoft/vcpkg/pull/28356), do you have any good suggestions for this issue? 
Thanks.

**Comment by quyykk** [comment]
Oh oops I messed up when I updated Boost, sorry. Fixed in #28504. I tested it with the sample code posted above

**Comment by YunchengLiu** [comment]
@quyykk @MonicaLiu0311
Thanks, waiting for an update!
